src/librustc_metadata/rmeta/decoder.rs

```diff
@@ -519,14 +519,13 @@ impl EntryKind {
             EntryKind::Enum(..) => DefKind::Enum,
             EntryKind::MacroDef(_) => DefKind::Macro(MacroKind::Bang),
             EntryKind::ForeignType => DefKind::ForeignTy,
-
-            EntryKind::ForeignMod
-            | EntryKind::GlobalAsm
-            | EntryKind::Impl(_)
-            | EntryKind::Field
-            | EntryKind::Generator(_)
-            | EntryKind::Closure => return None,
-        })
+            EntryKind::Impl(_) => DefKind::Impl,
+            EntryKind::Closure => DefKind::Closure,
+            EntryKind::ForeignMod => DefKind::ForeignMod,
+            EntryKind::GlobalAsm => DefKind::GlobalAsm,
+            EntryKind::Field => DefKind::Field,
+            EntryKind::Generator(_) => DefKind::Closure,
```

Comment:
Hmm should we have a `DefKind::Generator`? I forgot about that.

Reply:
I thought about that too. I think it would be preferable, but how would we tell from a `Node::Expr` if something is a generator (i.e. in librustc_hir)?

Reply:
You have a `TyCtxt` so you probably have a way to check. @Zoxc would know more.

Reply:
You can tell if it's a generator from `ExprKind::Closure`, not sure if we should have a `DefKind::Generator` though.

Reply:
@Zoxc Sorry, I don't quite understand. How can you tell from `ExprKind::Closure`?

Reply:
The `Option<Movability>` field will be `Some` for generators.